ddl/column.go

```diff
@@ -1490,81 +1594,24 @@ func (w *worker) doModifyColumn(
-			// `alter table tableName modify column b int after b` will return ver,ErrColumnNotExists.
-			// Modified the type definition of 'null' to 'not null' before this, so rollback the job when an error occurs.
-			job.State = model.JobStateRollingback
-			return infoschema.ErrColumnNotExists.GenWithStackByArgs(oldCol.Name, tblInfo.Name)
-		}
-
-		relative := model.FindColumnInfo(tblInfo.Columns, pos.RelativeColumn.Name.L)
-		if relative == nil || relative.State != model.StatePublic {
-			job.State = model.JobStateRollingback
-			return infoschema.ErrColumnNotExists.GenWithStackByArgs(pos.RelativeColumn, tblInfo.Name)
-		}
-
-		if relative.Offset < oldPos {
-			newPos = relative.Offset + 1
-		} else {
-			newPos = relative.Offset
-		}
-	} else if pos.Tp == ast.ColumnPositionFirst {
-		newPos = 0
-	}
-
-	columnChanged := make(map[string]*model.ColumnInfo)
-	columnChanged[oldCol.Name.L] = newCol
-
-	if newPos == oldPos {
-		tblInfo.Columns[newPos] = newCol
-	} else {
-		cols := tblInfo.Columns
-
-		// Reorder columns in place.
-		if newPos < oldPos {
-			// ******** +(new) ****** -(old) ********
-			// [newPos:old-1] should shift one step to the right.
-			copy(cols[newPos+1:], cols[newPos:oldPos])
-		} else {
-			// ******** -(old) ****** +(new) ********
-			// [old+1:newPos] should shift one step to the left.
-			copy(cols[oldPos:], cols[oldPos+1:newPos+1])
-		}
-		cols[newPos] = newCol
-
-		for i, col := range tblInfo.Columns {
-			if col.Offset != i {
-				columnChanged[col.Name.L] = col
-				col.Offset = i
-			}
-		}
+	if pos != nil && pos.RelativeColumn != nil && oldCol.Name.L == pos.RelativeColumn.Name.L {
+		// `alter table tableName modify column b int after b` will return ver, ErrColumnNotExists.
+		// Modified the type definition of 'null' to 'not null' before this, so rollback the job when an error occurs.
+		job.State = model.JobStateRollingback
+		return infoschema.ErrColumnNotExists.GenWithStackByArgs(oldCol.Name, tblInfo.Name)
 	}
-
-	// Change offset and name in indices.
-	for _, idx := range tblInfo.Indices {
-		for _, c := range idx.Columns {
-			cName := c.Name.L
-			// With the unique changing column name, the format is designed as `_Col$_xx_uniqueNum`.
-			// The suffix `_uniqueNum` will be trimmed when we get the origin changing column name.
-			// There is a possibility that some other column named as `xx_xx_xx`, and it will be get
-			// trimmed as `xx_xx`. So here we check here and only do the trim for the changing index column.
-			if len(changingColUniqueLowerName) != 0 && c.Name.L == changingColUniqueLowerName {
-				cName = strings.ToLower(getChangingColumnOriginName(c))
-			}
-			if newCol, ok := columnChanged[cName]; ok {
-				c.Name = newCol.Name
-				c.Offset = newCol.Offset
-			}
-		}
+	destOffset, err := locateOffsetToMove(oldCol.Offset, pos, tblInfo)
+	if err != nil {
+		job.State = model.JobStateRollingback
+		return infoschema.ErrColumnNotExists.GenWithStackByArgs(oldCol.Name, tblInfo.Name)
```

Comment:
Do we replace it with `return errors.Trace(err)` like https://github.com/pingcap/tidb/pull/33631/files#diff-42e0180cdc8c14def674a9e45d908d6700f6a8ccba0f9ff4b14b27c0daa49859R1094?